src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1267,6 +1269,36 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
             GetParameterCounts(m1, arguments, out m1ParameterCount, out m1ParametersUsedIncludingExpansionAndOptional);
             GetParameterCounts(m2, arguments, out m2ParameterCount, out m2ParametersUsedIncludingExpansionAndOptional);
 
+            // We might have got out of the loop above early and allSame isn't completely calculated.
```

Comment:
I can't tell what this comment is trying to tell me. Which `if` statement do we need to not skip over?

I feel like this method has gotten too long to be understood on its own. Can it be refactored into a set of helper methods?